python/semantic_kernel/agents/open_ai/responses_agent_thread_actions.py

```diff
@@ -653,13 +669,25 @@ def _prepare_chat_history_for_request(
     ) -> Any:
         """Prepare the chat history for a request.
 
-        We must skip any items of type
-        AnnotationContent, StreamingAnnotationContent, FileReferenceContent,
-        or StreamingFileReferenceContent, and always map the role to either user,
-        assistant, or developer.
+        Uses a pass-through approach: raw output items from responses are passed
+        directly to the next request, while only converting user messages to input format.
+        This ensures all response items (including function calls and reasoning) are preserved intact.
         """
         response_inputs: list[Any] = []  # type: ignore
+
         for message in chat_history.messages:
+            # Check if this message has raw output items from a response
+            if (
+                hasattr(message, "inner_content")
+                and message.inner_content
+                and hasattr(message.inner_content, "output")
+                and message.inner_content.output
+            ):
+                # Pass through raw output items directly to preserve API compliance
+                response_inputs.extend(message.inner_content.output)
```

Comment:
Not sure I follow why this inner_content / inner_content.output attribute check was introduced.

Reply:
This was a pass through to fix function calling, but u resolved this better with PR #12907.